### PR TITLE
Typescript support

### PR DESCRIPTION
**Pull request by chrmod**
Fix #

## Checklist:

- [ ] My squashed commit messages follows the [7 golden rules](https://chris.beams.io/posts/git-commit/)
- [ ] I updated or created necessary unit tests
- [ ] I've tested my changes in Dark Mode, iPad, iOS <= 12, Landscape and in German

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  root: true,
+  extends: ['@react-native-community', 'airbnb-typescript', 'prettier', 'prettier/@typescript-eslint', 'prettier/react'],
```
Comment (remusao) [inline .eslintrc.js]: Nitpick: having each on its own line would increase readability.

File: ReactNative/js/bridge-manager.js [hunk 2]
```diff
@@ -17,43 +18,44 @@ export default class BridgeManager {
   }
 
   async onAction({ module, action, args, id }) {
-    for(const listener of this.actionListeners) {
+    [...this.actionListeners].some(listener => {
       try {
         const handled = listener({ module, action, args, id });
         if (handled) {
-          return;
+          return false;
```
Comment (remusao) [inline ReactNative/js/bridge-manager.js]: Shouldn't this be `true`? Meaning that the first time a listener handles an action you stop.

File: ReactNative/js/components/ListItem.jsx [hunk 3]
```diff
@@ -42,6 +39,7 @@ export default function ListItem({ url, title, onPress, label }) {
 
   const name = parse(url).domain;
 
+  /* eslint-disable prettier/prettier */
```
Comment (remusao) [inline ReactNative/js/components/ListItem.jsx]: This will only disable eslint, not prettier formatting right? So if you run prettier it will still reformat this part (unless prettier also understands eslint directives).

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "isolatedModules": true,
```
Comment (remusao) [inline tsconfig.json]: Out of curiosity, why is this needed?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    "jsx": "react",
+    "lib": ["es6"],
```
Comment (remusao) [inline tsconfig.json]: Is this really needed if you use `target` `esnext`?

File: package.json [hunk 6]
```diff
@@ -8,6 +8,7 @@
     "build-detox": "detox build",
     "build-user-scripts": "webpack --config webpack.config.js",
     "test-detox": "detox test --take-screenshots all",
+    "lint": "tsc --noEmit && eslint --ext .js,.jsx,.ts,.tsx ./ReactNative/js",
```
Comment (remusao) [inline package.json]: What about a command to run `prettier`? I use the following commands in multiple projects:

```json
    "format:check": "prettier '{src,test}/**/*.ts' --check",
    "format:fix": "prettier '{src,test}/**/*.ts' --write",
    "lint:check": "tslint --config ./tslint.json --project ./tsconfig.json",
    "lint:fix": "tslint --config ./tslint.json --project ./tsconfig.json --fix",
    "typecheck": "tsc --project ./tsconfig.json",
    "typecheck:watch": "tsc --project ./tsconfig.json --watch",
    "yolo": "npm run format:fix && npm run lint:fix && npm run typecheck",
```

File: ReactNative/js/components/ListItem.jsx (hunk 3 above)
Comment (chrmod) [reply ReactNative/js/components/ListItem.jsx]: IDK, this suppresses the VS Code errors. We don't enforce automatic prettier.

File: tsconfig.json (hunk 4 above)
Comment (chrmod) [reply tsconfig.json]: I've copied this from some blog post. Will remove it now.

File: tsconfig.json (hunk 5 above)
Comment (chrmod) [reply tsconfig.json]: I have no clue. What would you suggest?

File: package.json (hunk 6 above)
Comment (chrmod) [reply package.json]: yolo 😂

Let's address this in separate PR.

Comment (remusao) [reply package.json]: Life is too short to run multiple commands :P

File: tsconfig.json (hunk 5 above)
Comment (remusao) [reply tsconfig.json]: I usually leave `target: "es6"` and it defaults to importing all the necessary libs.
